cmd/promtool/tsdb.go

```diff
@@ -398,21 +398,24 @@ func openBlock(path, blockID string) (*tsdb.DBReadOnly, tsdb.BlockReader, error)
 	if err != nil {
 		return nil, nil, err
 	}
-	blocks, err := db.Blocks()
-	if err != nil {
-		return nil, nil, err
-	}
+
 	var block tsdb.BlockReader
+
+	if blockID == "" {
+		blockID, err = db.LastBlockID(nil)
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+
 	if blockID != "" {
-		for _, b := range blocks {
-			if b.Meta().ULID.String() == blockID {
-				block = b
-				break
-			}
+		b, err := db.Block(nil, filepath.Join(blockID))
+		if err != nil {
+			return nil, nil, err
 		}
-	} else if len(blocks) > 0 {
-		block = blocks[len(blocks)-1]
+		block = b
 	}
+
 	if block == nil {
```

Comment:
We will never hit this condition now. And if the block did not exist, the above `db.Block()` will just return an error. So to have the right error message, we should check if the block directory exists and if the block ID parses the ulid.ParseStrict before we try to open the block.

Reply:
Addressed